src/Raven.Server/Rachis/Elector.cs

```diff
@@ -54,12 +54,25 @@ public void HandleVoteRequest()
 
                         if (clusterTopology.TopologyId == null)
                         {
-                            _connection.Send(context, new RequestVoteResponse
+                            if (HasAlreadyVoted(context, rv, out whoGotMyVoteIn))
                             {
-                                Term = rv.Term,
-                                VoteGranted = true,
-                                Message = "I might vote for you, because I'm not part of any cluster."
-                            });
+                                _connection.Send(context, new RequestVoteResponse
+                                {
+                                    Term = _engine.CurrentTerm,
+                                    VoteGranted = false,
+                                    Message = $"Already voted in {rv.LastLogTerm}, for {whoGotMyVoteIn}"
+                                });
+                            }
+                            else
+                            {
+                                _connection.Send(context, new RequestVoteResponse
```

Comment:
And where are you recording that you voted for him in this term?

Reply:
this is trial elections